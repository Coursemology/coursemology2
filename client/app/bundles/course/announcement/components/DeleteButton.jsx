import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  onClick: PropTypes.func.isRequired,
};

class DeleteButton extends React.Component {
  render() {
    return (
      <span className="btn btn-danger danger" title="Delete Announcement" onClick={() => this.props.onClick()}>
        <i className="fa fa-trash" />
      </span>
    );
  }
}

DeleteButton.propTypes = propTypes;

export default DeleteButton;
