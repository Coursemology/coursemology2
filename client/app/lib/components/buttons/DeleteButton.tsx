import { useState } from 'react';
import { IconButton, IconButtonProps, Tooltip } from '@mui/material';
import Delete from '@mui/icons-material/Delete';
import ConfirmationDialog from '../ConfirmationDialog';

interface Props extends IconButtonProps {
  disabled: boolean;
  onClick: () => Promise<void>;
  confirmMessage?: string;
  tooltip?: string;
}

const DeleteButton = ({
  disabled,
  onClick,
  confirmMessage,
  tooltip = '',
  ...props
}: Props): JSX.Element => {
  const [dialogOpen, setDialogOpen] = useState(false);

  return (
    <>
      <Tooltip title={tooltip}>
        <span>
          <IconButton
            disabled={disabled}
            onClick={(): void => {
              if (confirmMessage) {
                setDialogOpen(true);
              } else {
                onClick();
              }
            }}
            color="error"
            {...props}
          >
            <Delete />
          </IconButton>
        </span>
      </Tooltip>
      {dialogOpen && (
        <ConfirmationDialog
          message={confirmMessage}
          disableCancelButton={disabled}
          disableConfirmButton={disabled}
          open={dialogOpen}
          onCancel={(): void => setDialogOpen(false)}
          onConfirm={(): void => {
            onClick().finally(() => setDialogOpen(false));
          }}
        />
      )}
    </>
  );
};

export default DeleteButton;
