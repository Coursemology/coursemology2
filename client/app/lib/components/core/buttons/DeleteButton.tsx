import { ReactNode, useState } from 'react';
import Delete from '@mui/icons-material/Delete';
import { IconButton, IconButtonProps, Tooltip } from '@mui/material';

import useTranslation from 'lib/hooks/useTranslation';
import formTranslations from 'lib/translations/form';

import Prompt from '../dialogs/Prompt';

interface DeleteButtonProps extends IconButtonProps {
  disabled: boolean;
  onClick: () => Promise<void>;
  confirmMessage?: ReactNode;
  children?: ReactNode;
  tooltip?: string;
  loading?: boolean;
  title?: string;
  confirmLabel?: string;
}

const DeleteButton = (props: DeleteButtonProps): JSX.Element => {
  const {
    disabled,
    onClick,
    confirmMessage,
    children,
    tooltip,
    loading,
    title,
    confirmLabel,
    ...otherProps
  } = props;
  const { t } = useTranslation();
  const [dialogOpen, setDialogOpen] = useState(false);
  const promptContents = confirmMessage ?? children;

  return (
    <>
      <Tooltip title={tooltip ?? t(formTranslations.delete)}>
        <span>
          <IconButton
            color="error"
            data-testid="DeleteIconButton"
            disabled={disabled}
            {...otherProps}
            onClick={(): void => {
              if (promptContents) {
                setDialogOpen(true);
              } else {
                props.onClick();
              }
            }}
          >
            <Delete data-testid="DeleteIcon" />
          </IconButton>
        </span>
      </Tooltip>

      <Prompt
        contentClassName="space-y-4"
        disabled={loading ?? disabled}
        onClickPrimary={(): void => {
          onClick().finally(() => setDialogOpen(false));
        }}
        onClose={(): void => setDialogOpen(false)}
        open={dialogOpen}
        primaryColor="error"
        primaryLabel={confirmLabel ?? t(formTranslations.delete)}
        title={title}
      >
        {promptContents}
      </Prompt>
    </>
  );
};

export default DeleteButton;
