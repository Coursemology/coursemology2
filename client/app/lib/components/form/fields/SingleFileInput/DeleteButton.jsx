import { Component } from 'react';
import { FormattedMessage } from 'react-intl';
import Close from '@mui/icons-material/Close';
import { Badge, IconButton, Tooltip } from '@mui/material';
import PropTypes from 'prop-types';

import translations from './translations';

const styles = {
  badge: {
    position: 'absolute',
    top: 0,
    right: 0,
  },
};

export default class DeleteButton extends Component {
  renderIcon() {
    return (
      <Tooltip title={<FormattedMessage {...translations.removeFile} />}>
        <span>
          <IconButton
            onClick={this.props.handleCancel}
            size="small"
            style={styles.badgeStyle}
          >
            <Close />
          </IconButton>
        </span>
      </Tooltip>
    );
  }

  render() {
    return <Badge style={styles.badge}>{this.renderIcon()}</Badge>;
  }
}

DeleteButton.propTypes = {
  handleCancel: PropTypes.func,
};
