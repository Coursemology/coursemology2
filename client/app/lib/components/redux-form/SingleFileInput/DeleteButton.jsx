import { Component } from 'react';
import PropTypes from 'prop-types';
import { Badge, IconButton, Tooltip } from '@material-ui/core';
import { grey } from '@material-ui/core/colors';
import Close from '@material-ui/icons/Close';
import { FormattedMessage } from 'react-intl';
import translations from './translations';

const styles = {
  badge: {
    position: 'absolute',
    top: 0,
    right: 0,
  },
  badgeStyle: {
    backgroundColor: grey[400],
  },
};

export default class DeleteButton extends Component {
  renderIcon() {
    return (
      <Tooltip title={<FormattedMessage {...translations.removeFile} />}>
        <IconButton onClick={this.props.handleCancel} style={styles.badgeStyle}>
          <Close />
        </IconButton>
      </Tooltip>
    );
  }

  render() {
    return (
      <Badge
        //  badgeContent={this.renderIcon()}
        // badgeStyle={styles.badgeStyle}
        style={styles.badge}
      >
        {this.renderIcon()}
      </Badge>
    );
  }
}

DeleteButton.propTypes = {
  handleCancel: PropTypes.func,
};
