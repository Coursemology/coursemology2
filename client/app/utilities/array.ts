type StringKeys<T> = {
  [K in keyof T]: T[K] extends string | number | symbol ? K : never;
}[keyof T];

export const arrayToObjectWithKey = <
  T extends Record<StringKeys<T>, string | number | symbol>,
  TKeyName extends keyof Record<StringKeys<T>, string | number | symbol>,
>(
  array: T[],
  key: TKeyName,
): Record<T[TKeyName], T> =>
  Object.fromEntries(array.map((a) => [a[key], a])) as Record<T[TKeyName], T>;
