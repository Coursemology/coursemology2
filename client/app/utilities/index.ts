export const getIdFromUnknown = (id?: string | null): number | undefined =>
  parseInt(id ?? '', 10) || undefined;

export const getMailtoURLWithBody = (
  email: string,
  subject: string,
  body: string,
): string => {
  const encodedSubject = encodeURIComponent(subject);
  const encodedBody = encodeURIComponent(body);
  return `mailto:${email}?subject=${encodedSubject}&body=${encodedBody}`;
};
