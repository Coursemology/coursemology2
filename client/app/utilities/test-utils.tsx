// eslint-disable-next-line import/no-extraneous-dependencies
import { render, RenderResult } from '@testing-library/react';

import TestApp, { CustomRenderOptions } from './TestApp';

const customRender = (
  ui: JSX.Element,
  options?: CustomRenderOptions,
): RenderResult => render(<TestApp {...options}>{ui}</TestApp>);

// eslint-disable-next-line import/no-extraneous-dependencies
export * from '@testing-library/react';
export { customRender as render };
