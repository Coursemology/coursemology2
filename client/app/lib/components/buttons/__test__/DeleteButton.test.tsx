import { fireEvent, render, screen, RenderResult } from 'utilities/test-utils';
import DeleteButton from '../DeleteButton';

let documentBody: RenderResult;

describe('<DeleteButton />', () => {
  describe('when the delete button is rendered without confirmation dialog', () => {
    beforeEach(() => {
      documentBody = render(
        <DeleteButton disabled={false} onClick={jest.fn()} loading={false} />,
      );
    });

    it('shows the delete icon button', () => {
      expect(documentBody.getByTestId('DeleteIconButton')).toBeVisible();
      expect(documentBody.getByTestId('DeleteIcon')).toBeVisible();
    });

    it('does not show the confirmation dialog when clicked', () => {
      // Before clicking
      expect(documentBody.queryByTestId('ConfirmationDialog')).toBeNull();

      fireEvent.click(screen.getByTestId('DeleteIconButton'));
      // After clicking
      expect(documentBody.queryByTestId('ConfirmationDialog')).toBeNull();
    });

    it('matches the snapshot', () => {
      const { baseElement } = documentBody;
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('when the delete button is disabled', () => {
    beforeEach(() => {
      documentBody = render(
        <DeleteButton
          disabled
          onClick={jest.fn()}
          loading={false}
          confirmMessage="Are you sure you wish to delete?"
        />,
      );
    });

    it('shows the delete icon button', () => {
      expect(documentBody.getByTestId('DeleteIconButton')).toBeVisible();
      expect(documentBody.getByTestId('DeleteIcon')).toBeVisible();
      expect(documentBody.getByTestId('DeleteIconButton')).toBeDisabled();
    });
  });

  describe('when the delete button is rendered with confirmation dialog', () => {
    beforeEach(() => {
      documentBody = render(
        <DeleteButton
          disabled={false}
          onClick={jest.fn()}
          loading={false}
          confirmMessage="Are you sure you wish to delete?"
        />,
      );
    });

    it('shows the delete icon button', () => {
      expect(documentBody.getByTestId('DeleteIconButton')).toBeVisible();
      expect(documentBody.getByTestId('DeleteIcon')).toBeVisible();
    });

    it('shows the confirmation dialog when clicked', () => {
      // Before clicking delete button
      expect(documentBody.queryByTestId('ConfirmationDialog')).toBeNull();

      fireEvent.click(screen.getByTestId('DeleteIconButton'));
      // After clicking delete button
      expect(documentBody.getByTestId('ConfirmationDialog')).toBeVisible();

      expect(documentBody.getByText('Continue')).toBeEnabled();
      expect(documentBody.getByText('Cancel')).toBeEnabled();

      // After clicking cancel in the confirmation dialog, it is closed
      fireEvent.click(screen.getByText('Cancel'));
      expect(documentBody.queryByTestId('ConfirmationDialog')).toBeNull();
    });
  });
});
