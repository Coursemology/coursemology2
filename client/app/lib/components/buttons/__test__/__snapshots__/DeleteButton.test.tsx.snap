// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteButton /> when the delete button is rendered without confirmation dialog matches the snapshot 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1jo1f9u-MuiButtonBase-root-MuiIconButton-root"
      data-testid="DeleteIconButton"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
